### PR TITLE
feat: add ability reading stream_listpacks_2/3 rdb types

### PR DESCRIPTION
**Pull request by BorysTheDev**
fixes: #4159

### CODE REVIEW

File: src/server/rdb_load.cc [hunk 1]
```diff
@@ -1892,6 +1894,30 @@ auto RdbLoaderBase::ReadStreams() -> io::Result<OpaqueObj> {
...
+  } else {
+    /* During migration the offset can be initialized to the stream's
+     * length. At this point, we also don't care about tombstones
+     * because CG offsets will be later initialized as well. */
+    load_trace->stream_trace->max_deleted_entry_id.ms = 0;
+    load_trace->stream_trace->max_deleted_entry_id.seq = 0;
+    load_trace->stream_trace->entries_added = load_trace->stream_trace->stream_len;
+
+    /* Since the rax is already loaded, we can find the first entry's * ID. */
+    // TODO add implementation
```
Comment (romange) [inline src/server/rdb_load.cc:1917]: it's not clear from TODO what needs to be implemented. What are the the values we read but discarded. Please expand on it.

File: src/server/rdb_load.cc [hunk 2]
```diff
@@ -1913,6 +1939,24 @@ auto RdbLoaderBase::ReadStreams() -> io::Result<OpaqueObj> {
     SET_OR_UNEXPECT(LoadLen(nullptr), cgroup.ms);
     SET_OR_UNEXPECT(LoadLen(nullptr), cgroup.seq);
 
+    uint64_t cg_offset;
+    if (rdbtype >= RDB_TYPE_STREAM_LISTPACKS_2) {
+      SET_OR_UNEXPECT(LoadLen(nullptr), cg_offset);
+      (void)cg_offset;
+    } else {
+      // TODO implement
```
Comment (romange) [inline src/server/rdb_load.cc:1947]: this one is fine.

File: src/server/rdb_load.cc [hunk 3]
```diff
@@ -1958,6 +2002,13 @@ auto RdbLoaderBase::ReadStreams() -> io::Result<OpaqueObj> {
 
       SET_OR_UNEXPECT(FetchInt<int64_t>(), consumer.seen_time);
 
+      if (rdbtype >= RDB_TYPE_STREAM_LISTPACKS_3) {
+        SET_OR_UNEXPECT(FetchInt<int64_t>(), consumer.active_time);
+      } else {
+        /* That's the best estimate we got */
```
Comment (romange) [inline src/server/rdb_load.cc:2008]: does it also require improvement  or is it aligned with valkey 7.2?

File: src/server/rdb_test.cc [hunk 4]
```diff
@@ -655,4 +655,14 @@ TEST_F(RdbTest, SnapshotTooBig) {
   ASSERT_THAT(resp, ErrArg("Out of memory"));
 }
 
+TEST_F(RdbTest, LoadStream2) {
+  auto ec = LoadRdb("RDB_TYPE_STREAM_LISTPACKS_2.rdb");
```
Comment (romange) [inline src/server/rdb_test.cc:659]: please expand the tests to actually check that the streams are present and they have items.

File: src/server/rdb_load.cc (hunk 3 above)
Comment (BorysTheDev) [reply src/server/rdb_load.cc:2008]: It's a copy of Redis code and it's ok

**Pull request merged**